samples: spi_fujitsu_fram: fixed some logic problems

markus-gerber opened the pull request:
Fixed mb85rs64v_read_id as before it would call mb85rs64v_access with
"cmd = MB85RS64V_MANUFACTURER_ID_CMD". This would only call spi_write
instead of spi_transceive because cmd is neither WRITE_CMD nor READ_CMD.

One other bug in mb85rs64v_access was that if
"cmd = MB85RS64V_WRITE_ENABLE_CMD", bufs[0].len was 0 instead of 1 so
these command were not executed correctly.
This made write_bytes not really write bytes, as writing was still
disabled for the fram module.

Furthermore I set the cs property of spi_config, because it does not work
without setting it. I also included a sample overlay file how to setup
a spi device for the esp32c3_devkitm.

Kylof61034 commented:
This is my first pull request here, so if I got anything wrong or you have other questions, please contact me :)

markus-gerber committed:
samples:  spi_fujitsu_fram: fixed some logic problems

Fixed mb85rs64v_read_id as before it would call mb85rs64v_access with
"cmd = MB85RS64V_MANUFACTURER_ID_CMD". This would only call spi_write
instead of spi_transceive because cmd is neither WRITE_CMD nor READ_CMD.

One other bug in mb85rs64v_access was that if
"cmd = MB85RS64V_WRITE_ENABLE_CMD", bufs[0].len was 0 instead of 1 so
these command were not executed correctly.
This made write_bytes not really write bytes, as writing was still
disabled for the fram module.

Furthermore I set the cs property of spi_config, because it does not work
without setting it. I also included a sample overlay file how to setup
a spi device for the esp32c3_devkitm.

Signed-off-by: Markus Gerber <gerber.markus1@web.de>

Kylof61034 commented:
Maybe I should explain the reason for this pull request:

I was trying to get the SPI FRAM module from Fujitsu working with an ESP32C3.
However, the example always failed when reading and verifying the id of that fram module.

I looked at the code and saw at least two big logic problems with the sample (when reading the id and disabling the write protection).
With the changes of my PR, the example worked as expected (tho I don't see how this example could have worked in the first place).

One other thing I also needed to add to make this example run was to set the cs pin (cs property of the spi_config struct) and define an overlay file for my device.
I am not sure if there is another way to do that, but I also included it because I feel like there is not much documentation about that.
Let me know what you think about the last change!

So as I was really confused when running this example, I hope I can save someone else from this confusion with this PR :)

MaureenHelm commented:
@nashif please take a look
